src/main/java/Parser.java

```diff
@@ -0,0 +1,87 @@
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.Arrays;
+import java.util.List;
+
+public class Parser {
+    private static final List<String> DATE_FORMATS = Arrays.asList("yyyy/MM/dd HHmm", "y/M/d HHmm", "y-M-d HHmm");
+    public static Command parse(String userInput) throws DukeException {
+        int i = userInput.trim().indexOf(' ');
+        String command = userInput;
+        String detail = "";
+        if (i > 0) {
+            command = userInput.substring(0, i);
+            detail = userInput.substring(i).trim();
+        }
+        switch (command) {
+            case ("bye") :
+                return new ByeCommand();
+            case ("list") :
+                return new ListCommand();
+            case ("todo") :
+                return new AddCommand(parseTodo(detail));
+            case ("deadline") :
+                return new AddCommand(parseDeadline(detail));
+            case ("event") :
+                return new AddCommand(parseEvent(detail));
+            case ("delete") :
+                return new DeleteCommand(parseNumber(detail));
+            case ("done") :
+                return new DoneCommand(parseNumber(detail));
+            default:
+                throw new UnexpectedInputException();
```

Comment:
I don't think there is supposed to be indentation for case clauses for switch statements so perhaps you could configure your IDE to follow the style in the Java coding standards?